lock-api/src/test/java/com/palantir/lock/watch/ConjureLockRequestMetadataUtilsTest.java

```diff
@@ -0,0 +1,170 @@
+    private static final Map<LockDescriptor, ChangeMetadata> LOCKS_WITH_METADATA = ImmutableMap.of(
+            LOCK_1,
+            CHANGE_METADATA_LIST.get(0),
+            LOCK_2,
+            CHANGE_METADATA_LIST.get(1),
+            LOCK_3,
+            CHANGE_METADATA_LIST.get(2),
+            LOCK_4,
+            CHANGE_METADATA_LIST.get(3));
+    private static final LockRequestMetadata LOCK_REQUEST_METADATA = LockRequestMetadata.of(LOCKS_WITH_METADATA);
+    private static final Map<Integer, ConjureChangeMetadata> CONJURE_LOCKS_WITH_METADATA = ImmutableMap.of(
+            0,
+            ConjureChangeMetadata.unchanged(ConjureUnchangedChangeMetadata.of()),
+            1,
+            ConjureChangeMetadata.updated(
+                    ConjureUpdatedChangeMetadata.of(Bytes.from(BYTES_OLD), Bytes.from(BYTES_NEW))),
+            2,
+            ConjureChangeMetadata.deleted(ConjureDeletedChangeMetadata.of(Bytes.from(BYTES_DELETED))),
+            3,
+            ConjureChangeMetadata.created(ConjureCreatedChangeMetadata.of(Bytes.from(BYTES_CREATED))));
+
+    private static ConjureMetadataConversionResult conjureMetadataConversionResult;
+
+    // This is a good candidate for a static construction method, but we would rather avoid testing internals of
+    // IndexEncodingUtils (checksum computation) within the tests for Conjure conversion.
+    @BeforeClass
+    public static void setup() {
+        KeyListChecksum checksum =
+                IndexEncodingUtils.computeChecksum(ConjureLockRequestMetadataUtils.DEFAULT_CHECKSUM_TYPE, LOCK_LIST);
+        ConjureLockRequestMetadata conjureMetadata = ConjureLockRequestMetadata.builder()
+                .indexToChangeMetadata(CONJURE_LOCKS_WITH_METADATA)
+                .lockListChecksum(
+                        ConjureLockDescriptorListChecksum.of(checksum.type().getId(), Bytes.from(checksum.value())))
+                .build();
+        conjureMetadataConversionResult = ConjureMetadataConversionResult.builder()
+                .lockList(LOCK_LIST)
+                .conjureMetadata(conjureMetadata)
+                .build();
+    }
+
+    @Test
+    public void convertsToConjureCorrectly() {
+        // ImmutableSet remembers insertion order, which is critical here
+        assertThat(ConjureLockRequestMetadataUtils.toConjureIndexEncoded(
+                        ImmutableSet.copyOf(LOCK_LIST), LOCK_REQUEST_METADATA))
+                .isEqualTo(conjureMetadataConversionResult);
+    }
+
+    @Test
+    public void convertsFromConjureCorrectly() {
+        assertThat(ConjureLockRequestMetadataUtils.fromConjureIndexEncoded(conjureMetadataConversionResult))
+                .isEqualTo(LOCK_REQUEST_METADATA);
+    }
+
+    @Test
+    public void convertingToAndFromConjureYieldsOriginalForRandomData() {
+        int randSeed = (int) System.currentTimeMillis();
+        Random random = new Random(randSeed);
+        Set<LockDescriptor> lockDescriptors = Stream.generate(UUID::randomUUID)
```

Comment:
It's probably worth pulling this out into it's own separate method. Also the question I'd ask here, is random value generation really needed, or could we do this more deterministically? As presently the randomUUID does not use the same seed.

Reply:
See comment at the end. I mainly want to test sparse metadata and removed the randomness

Reply:
Given that I removed the randomness, is it fine to keep the data setup as part of the test or do you still think it should have its own method?